packages/api/src/external/ehr/athenahealth/command/process-patients-from-appointments-sub.ts

```diff
@@ -1,103 +1,98 @@
 import { getCxMappingsBySource } from "../../../../command/mapping/cx";
-import { getSecretValueOrFail } from "@metriport/core/external/aws/secret-manager";
-import { Config } from "../../../../shared/config";
-import { getPatientIdOrFail as singleGetPatientIdOrFail } from "./get-patient";
+import { EhrSources, getLookackTimeRange } from "../../shared";
+import { getPatientIdOrFail } from "./get-patient";
+import { getAthenaEnv } from "../shared";
 
 dayjs.extend(duration);
 
 const delayBetweenPracticeBatches = dayjs.duration(30, "seconds");
-const lastModifiedHoursLookback = 12;
+const catupUpLookback = dayjs.duration(12, "hours");
 const parallelPractices = 10;
 const parallelPatients = 2;
 
-const region = Config.getAWSRegion();
-const athenaEnvironment = Config.getAthenaHealthEnv();
-const athenaClientKeySecretArn = Config.getAthenaHealthClientKeyArn();
-const athenaClientSecretSecretArn = Config.getAthenaHealthClientSecretArn();
-
 type PatientAppointment = {
   cxId: string;
   athenaPracticeId: string;
   athenaPatientId: string;
 };
 
-export async function getPatientIdsOrFailFromAppointmentsSub({
-  catchUp,
-}: {
-  catchUp: boolean;
-}): Promise<void> {
-  const { log } = out(`AthenaHealth getPatientIdsOrFailFromAppointmentsSub`);
-  if (!athenaEnvironment || !athenaClientKeySecretArn || !athenaClientSecretSecretArn) {
-    throw new Error("AthenaHealth not setup");
-  }
-  const cxMappings = await getCxMappingsBySource({ source: EhrSources.athena });
+export async function processPatientsFromAppointmentsSub({ catchUp }: { catchUp: boolean }) {
+  const { log } = out(`AthenaHealth getPatientIdsOrFailFromAppointmentsSub - catchUp: ${catchUp}`);
+  const { environment, clientKey, clientSecret } = await getAthenaEnv();
 
-  const athenaClientKey = await getSecretValueOrFail(athenaClientKeySecretArn, region);
-  const athenaClientSecret = await getSecretValueOrFail(athenaClientSecretSecretArn, region);
+  const { startRange, endRange } = catchUp
+    ? getLookackTimeRange({ lookback: catupUpLookback })
+    : {
+        startRange: undefined,
+        endRange: undefined,
+      };
+  if (startRange || endRange) log(`Getting appointments from ${startRange} to ${endRange}`);
 
-  const currentDatetime = buildDayjs(new Date());
-  const startLastModifiedDate = buildDayjs(currentDatetime)
-    .hour(currentDatetime.hour() - lastModifiedHoursLookback)
-    .toDate();
-  const endLastModifiedDate = buildDayjs(currentDatetime).toDate();
-  if (catchUp) {
-    log(`Getting appointments from ${startLastModifiedDate} to ${endLastModifiedDate}`);
-  } else {
-    log(`Getting appointments since last run`);
-  }
+  const cxMappings = await getCxMappingsBySource({ source: EhrSources.athena });
 
-  const patientAppointments: PatientAppointment[] = [];
-  const getAppointmentsErrors: string[] = [];
-  const getAppointmentsArgs = cxMappings.map(mapping => {
+  const patientAppointmentsFromGetAppointmentsFromSubByPractice: PatientAppointment[] = [];
```

Comment:
🤯 That's a complicated name

Reply:
Taken to slack. I'd argue it's not complicated, just long.

Reply:
it's an array of patient appointments from the getAppointsmentsFromSubByPractice method :)

Reply:
No longer relevant.